java-util/src/main/java/io/druid/java/util/common/parsers/AbstractFlatTextFormatParser.java

```diff
@@ -177,13 +156,50 @@ public void setFieldNames(final String header)
         setFieldNames(ParserUtils.generateFieldNames(values.size()));
       }
 
-      Map<String, String> fieldValueMap = Utils.zipMapPartial(fieldNames, values);
-      return Maps.transformEntries(fieldValueMap, multiValueTransformer);
+      return computeKeyValues(fieldNames, values);
     }
     catch (Exception e) {
       throw new ParseException(e, "Unable to parse row [%s]", input);
     }
   }
 
+  private Map<String, Object> computeKeyValues(List<String> fieldNames, List<String> values)
+  {
+    Map<String, Object> retVal = new LinkedHashMap<>();
+    Iterator<String> keysIter = fieldNames.iterator();
```

Comment:
Could be simpler for-each loop over keys

Reply:
```
private Map<String, Object> computeKeyValues(List<String> fieldNames, List<String> values)
  {
    Map<String, Object> retVal = new LinkedHashMap<>();
    Iterator<String> valsIter = values.iterator();

    for (String key: fieldNames) {
      if (valsIter.hasNext()) {
        Object splitedValue = spliteValue(key, valsIter.next());
        retVal.put(key, splitedValue);
      } else {
        break;
      }
    }
    return retVal;
  }
```
Above is better ?